### PR TITLE
core(stack-packs): move to lighthouse-stack-packs npm package

### PR DESCRIPTION
**Pull request by connorjclark**
cont. https://github.com/GoogleChrome/lighthouse/pull/10138

This also includes an i18n import.

### CODE REVIEW

**Comment by wardpeet** [comment]
Thanks @connorjclark for taking it over!

File: proto/sample_v2_round_trip.json [hunk 1]
```diff
@@ -0,0 +1,5497 @@
+{
```
Comment (patrickhulce) [inline proto/sample_v2_round_trip.json:1]: didn't we get rid of this? revert?

File: lighthouse-core/lib/i18n/locales/zh.json [hunk 2]
```diff
@@ -383,6 +383,15 @@
   "lighthouse-core/audits/apple-touch-icon.js | title": {
     "message": "提供了有效的 `apple-touch-icon`"
   },
+  "lighthouse-core/audits/autocomplete.js | description": {
```
Comment (patrickhulce) [inline lighthouse-core/lib/i18n/locales/zh.json:386]: this seems unrelated and perhaps out of date. was this updated with old scripts?

**Review by patrickhulce — CHANGES_REQUESTED** [review]
impl LGTM but lots of generated files to clean up it seems

File: lighthouse-core/lib/i18n/locales/zh.json (hunk 2 above)
Comment (patrickhulce) [reply lighthouse-core/lib/i18n/locales/zh.json:386]: ah was this a result of "This also includes an i18n import."?

File: lighthouse-core/lib/stack-packs.js [hunk 3]
```diff
@@ -5,13 +5,26 @@
...
+  }
+
+  return require.resolve(module);
+}
 
 /**
  * Pairs consisting of a stack pack's ID and the set of stacks needed to be
  * detected in a page to display that pack's advice.
- * @type {Array<{packId: string, requiredStacks: Array<string>}>}
+ * @type {Array<{packId: string, requiredStacks: string[]}>}
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply lighthouse-core/lib/stack-packs.js:27]: in my defense, I thought this was new code

File: lighthouse-core/lib/i18n/locales/zh.json (hunk 2 above)
Comment (connorjclark) [reply lighthouse-core/lib/i18n/locales/zh.json:386]: Not simply. For the new keys (but old strings) to be in all the locales, an i18n roundtrip was necessary.

Comment (connorjclark) [reply lighthouse-core/lib/i18n/locales/zh.json:386]: It's not out of date, it's just the string is very new and hasn't been translated yet.

File: proto/sample_v2_round_trip.json (hunk 1 above)
Comment (connorjclark) [reply proto/sample_v2_round_trip.json:1]: ahhhhhhh 🔥

File: lighthouse-core/lib/i18n/locales/zh.json (hunk 2 above)
Comment (patrickhulce) [reply lighthouse-core/lib/i18n/locales/zh.json:386]: I gotcha, you did an i18n roundtrip to preserve the strings that got moved in this PR and the addition of these unrelated strings is a side effect 👍

File: lighthouse-core/lib/stack-packs.js (hunk 3 above)
Comment (connorjclark) [inline lighthouse-core/lib/stack-packs.js:27]:
```suggestion
 * @type {Array<{packId: string, requiredStacks: Array<string>}>}
```

**Review by patrickhulce — APPROVED** [review]

**Comment by patrickhulce** [comment]
@googlebot I consent.

**Comment by patrickhulce** [comment]
@connorjclark maybe it's waiting on your consent for some reason?

**Comment by connorjclark** [comment]
It's blocked on ward for some reason, but he's clearly given his consent so.

**Comment by brendankenny** [comment]
Looks like this broke stack pack display in the report, at least with sample_v2.

e.g. check out "Eliminate render-blocking resources" in https://lighthouse-627juwfex.vercel.app/english/ vs last report on master: https://lighthouse-hpoqd481r.vercel.app/english/

I assume it's from whatever switched from hyphens to underscores so audit lookup by id wouldn't work anymore.

https://github.com/GoogleChrome/lighthouse/blob/74bb627289d1ebd19d7b51fcd1209bf47e4bdfdf/lighthouse-core/test/results/sample_v2.json#L7992-L8007

**Comment by connorjclark** [comment]
they were a mix of `_` and `-` before the change to `.js`, and with that change they were all switched to `_` :( https://github.com/GoogleChrome/lighthouse-stack-packs/pull/38/files#diff-e4eccfef3eeb455a79349977b0518e71L2

I'll open a PR there.

**Comment by brendankenny** [comment]
looks like we don't have any renderer tests (especially using real data like `sample_v2`) either :/

**Comment by patrickhulce** [comment]
@connorjclark should we revert this and/or add a conversion+test in LH instead if it involves a PR + publish there?
